c++/src/Reader.cc

```diff
@@ -1108,13 +1107,75 @@ namespace orc {
     // internal methods
     proto::StripeFooter getStripeFooter(const proto::StripeInformation& info);
     void startNextStripe();
-    void checkOrcVersion();
     void selectType(const Type& type);
-    void readMetadata() const;
     void updateSelected(const std::list<uint64_t>& fieldIds);
     void updateSelected(const std::list<std::string>& fieldNames);
 
   public:
+   /**
+    * Constructor that lets the user specify additional options.
+    * @param filereader the object to read from
+    * @param options options for reading
+    */
+    RowReaderImpl(const ReaderImpl* filereader,
```

Comment:
We will have to extend the `Reader` interface with a notion of `metadata`, `ReaderOptions` etc. to achieve this. I am not sure what the right design choice is here.